### PR TITLE
extend the score process to support the normalizeScore and scoreWeight

### PR DESCRIPTION
**Pull request by huone1**
Signed-off-by: huone1 <huwanxing@huawei.com>

**What type of PR is this?**
/kind feature
**What this PR does / why we need it**:
extand the ScorePlugin and add NormalizeScore interface and support the ScoreWeight
**Which issue(s) this PR fixes**:
a part of #1330

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
LGTM. @Garrybest would you like to take a look?

File: pkg/scheduler/framework/interface.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ type Framework interface {
 	RunFilterPlugins(ctx context.Context, placement *policyv1alpha1.Placement, resource *workv1alpha2.ObjectReference, clusterv1alpha1 *clusterv1alpha1.Cluster) *Result
 
 	// RunScorePlugins runs the set of configured Score plugins, it returns a map of plugin name to cores
-	RunScorePlugins(ctx context.Context, placement *policyv1alpha1.Placement, clusters []*clusterv1alpha1.Cluster) (PluginToClusterScores, error)
+	RunScorePlugins(ctx context.Context, placement *policyv1alpha1.Placement, spec *workv1alpha2.ResourceBindingSpec, clusters []*clusterv1alpha1.Cluster) (PluginToClusterScores, error)
```
Comment (Garrybest) [inline pkg/scheduler/framework/interface.go:22]: Seems we haven't use `spec` for scoring. Is this parameter introduced for future work?

Comment (huone1) [reply pkg/scheduler/framework/interface.go:22]: Yeah, you guessed right.

**Comment by Garrybest** [comment]
Nice work!
/lgtm
/approve

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**